server.go

```diff
@@ -1212,6 +1214,30 @@ func newServer(listenAddrs []net.Addr, chanDB *channeldb.DB,
 		}
 	}
 
+	if len(cfg.ExternalHosts) != 0 {
+		s.hostAnn = netann.NewHostAnnouncer(netann.HostAnnouncerConfig{
+			Hosts:           cfg.ExternalHosts,
+			RefreshInterval: time.Minute * 5,
+			LookupHost: func(host string) (net.Addr, error) {
+				return lncfg.ParseAddressString(
+					host, strconv.Itoa(defaultPeerPort),
+					cfg.net.ResolveTCPAddr,
+				)
+			},
+			AnnounceNewIPs: func(addrs []net.Addr) error {
+				_, err := s.genNodeAnnouncement(
+					true, func(currentNodeAnn *lnwire.NodeAnnouncement) {
+						// TODO(roasbeef): would
+						// clobber any other ones or
+						// nat addrrs
+						currentNodeAnn.Addresses = addrs
```

Comment:
We could pass in the old addresses to `AnnounceNewIPs` and filter them out here.

Reply:
Implemented.